src/ynEIGEN/TokenStakingNode.sol

```diff
@@ -135,105 +143,132 @@ contract TokenStakingNode is
 
     /**
      * @notice Queues withdrawals for a specific strategy
-     * @param _strategy The strategy from which to withdraw
-     * @param _shares The number of shares to withdraw
-     * @return _fullWithdrawalRoots An array of withdrawal roots generated by the queueWithdrawals operation
+     * @param _strategy The strategy to withdraw from.
+     * @param _depositShares The number of deposit shares to withdraw.
+     * @return fullWithdrawalRoots An array of containing the withdrawal root of the queued withdrawal.
      */
     function queueWithdrawals(
         IStrategy _strategy,
-        uint256 _shares
-    ) external onlyTokenStakingNodesWithdrawer returns (bytes32[] memory _fullWithdrawalRoots) {
-
-        IStrategy[] memory _strategiesArray = new IStrategy[](1);
-        _strategiesArray[0] = _strategy;
-        uint256[] memory _sharesArray = new uint256[](1);
-        _sharesArray[0] = _shares;
-        IDelegationManagerTypes.QueuedWithdrawalParams[] memory _params = new IDelegationManagerTypes.QueuedWithdrawalParams[](1);
-        _params[0] = IDelegationManagerTypes.QueuedWithdrawalParams({
-            strategies: _strategiesArray,
-            depositShares: _sharesArray,
-            __deprecated_withdrawer: address(this)
+        uint256 _depositShares
+    ) external onlyTokenStakingNodesWithdrawer returns (bytes32[] memory fullWithdrawalRoots) {
+        IDelegationManagerTypes.QueuedWithdrawalParams[] memory queueWithdrawalParams = new IDelegationManagerTypes.QueuedWithdrawalParams[](1);
+
+        queueWithdrawalParams[0] = IDelegationManagerTypes.QueuedWithdrawalParams({
+            strategies: new IStrategy[](1),
+            depositShares: new uint256[](1),
+            __deprecated_withdrawer: address(0) // This field is ignored by EigenLayer v1.0.3.
         });
 
-        queuedShares[_strategy] += _shares;
+        queueWithdrawalParams[0].strategies[0] = _strategy;
+        queueWithdrawalParams[0].depositShares[0] = _depositShares;
+
+        IDelegationManagerWithAllocationManager delegationManager = IDelegationManagerWithAllocationManager(address(tokenStakingNodesManager.delegationManager()));
+
+        DepositScalingFactor memory dsf = DepositScalingFactor({_scalingFactor: delegationManager.depositScalingFactor(address(this), _strategy)});
+        uint256 maxMagnitude = delegationManager.allocationManager().getMaxMagnitude(delegationManager.delegatedTo(address(this)), _strategy);
+        uint256 withdrawableShares = SlashingLib.calcWithdrawable(dsf, _depositShares, maxMagnitude);
```

Comment:
In the implementation it uses slashingFactor instead of maxMagnitude. However, for NON beacon chain strategies, slashingFactor == maxMagnitude while for beacon chain strats, slashing factor == maxMagnitude * beaconChainScalingFactor